src/algebra/char_p/two.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import algebra.char_p.basic
+
+/-!
+# Lemmas about rings of characteristic two
+
+This file contains results about `char_p R 2`, in the `char_two` namespace.
+-/
+
+variables {R ι : Type*}
+
+namespace char_two
+
+lemma two_eq_zero [semiring R] [char_p R 2] : (2 : R) = 0 :=
+by rw [← nat.cast_two, char_p.cast_eq_zero]
+
+lemma bit0_eq_zero [semiring R] [char_p R 2] (x : R) : (bit0 x : R) = 0 :=
+by rw [bit0, ←two_smul R x, two_eq_zero, zero_smul]
+
+lemma neg_eq [ring R] [char_p R 2] (x : R) : -x = x :=
+by rw [neg_eq_iff_add_eq_zero, ←two_smul R x, two_eq_zero, zero_smul]
+
+lemma sub_eq_add [ring R] [char_p R 2] (x y : R) : x - y = x + y :=
+by rw [sub_eq_add_neg, neg_eq]
+
+lemma sub_eq_add' [ring R] [char_p R 2] : has_sub.sub = ((+) : R → R → R) :=
+funext $ λ x, funext $ λ y, sub_eq_add x y
+
+lemma add_sq [comm_semiring R] [char_p R 2] (x y : R) :
+  (x + y) ^ 2 = x ^ 2 + y ^ 2 :=
+begin
+  letI := fact.mk nat.prime_two,
```

Comment:
You can use `local attribute [instance] nat.fact_prime_two` near the top of the `char_two` namespace.

Reply:
Yep, I was waiting on the dependent PR in order to do that.